### ISSUE TITLE
[BUG]: Streamdeck remains active even if computer is locked

### ISSUE THREAD

**Issue by celgit** [streamdeck-linux-gui/streamdeck-linux-gui#23]
### In raising this issue I confirm that

- [X] I have fully completed the issue template
- [X] I have searched open and closed issues for duplicates
- [X] I have read the [Contribution Guidelines](https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/contributing/contributing-guide/)
- [X] I have read the [Code of Conduct](https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/contributing/code-of-conduct/)
- [X] I have read the [Documentation](https://streamdeck-linux-gui.github.io/streamdeck-linux-gui/)

### Describe the bug

If you lock your desktop the stream deck still remains active and you can still use it, while the computer remains locked.
(this was reported on the old github but got a comment to add the issue here if it still remains).

### Steps to reproduce the behavior

1. Add a button in streamdeck-ui that does something like change audio-output or similar, something that can be confirmed to work without a display. (I used audio-device-switching and play/pause media)
2. Lock your desktop
3. Press the buttons, for instance:
3a. Unpause media
3b. switch between headphones and speakers

### Expected behavior

Either it should turn the streamdeck off when desktop is locked or the buttons shouldn't do anything (alt 1 is better if possible).

### Screenshots

N/A, can film if needed.

### System Information

OS: Nobara Linux 38
Device: Streamdeck XL
Software:
A Linux compatible UI for the Elgato Stream Deck.
Version 2.0.15

streamdeck 0.9.3
pyside6 6.5.1.1
pillow 9.5.0
pynput 1.7.6

### Stream Deck Version

Streamdeck XL

**Comment by AWDDude** [comment]
I too would like this feature.

**Comment by sftim** [comment]
Also see https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/15

**Comment by Data-007** [comment]
This is an oldie but a goody 
See Timothy crosley  github
 You are able to press buttons (and therefore run commands) on computer that has been locked #282 
HIH

**Comment by sftim** [comment]
Related: https://github.com/timothycrosley/streamdeck-ui/pull/235

**Comment by StormyIceLeopard** [comment]
This should get high priority since it can be a security issue.

**Comment by 4s3ti** [comment]
> This should get high priority since it can be a security issue.
 
 Got the development branch ready to get that pull request.

**Comment by StormyIceLeopard** [comment]
> > This should get high priority since it can be a security issue.
> 
> Got the development branch ready to get your pull request.

I would contribute code wise if I could. But unfortunately that's beyond me.

**Comment by AlexandreBonneau** [comment]
Just my 0.02€, but some DE allows for controlling the play/pause status as well as the music volume of a locked session (for instance KDE does this).
This is what you want in most case because you don't want your colleagues to have to listen to high volume sounds while you are away.

That said this can be done on the computer directly using the keyboard/mouse, so disabling the Streamdeck altogether might be the best solution here in my opinion.

**Comment by pkoerner81929** [comment]
I had the same behaviour on Fedora 38 and created a solution. It works for screensavee locks or manual locks with Super-L.  Currently not supporting log-off from session. 


1. Run streamdeck-linux-gui as user service to allow start/stop (see installation guide)
2. Create another user systemd service, that will monitor the dbus for session events. 

Keep in mind to use special options to control user service as root:
systemctl --user --machine=<user>@.host <command> <service>


/usr/local/bin/watch_session.sh  (Edit with your username)

```
#!/bin/bash

dbus-monitor --session "type='signal',interface='org.gnome.ScreenSaver'" | \
while read x; do
    case "$x" in 
        *"boolean true"*) 
            # Screen locked
	    systemctl --user --machine=[username]@.host stop streamdeck.service;;
        *"boolean false"*) 
            # Screen unlocked
            systemctl --user --machine=[username]pkoerner@.host start streamdeck.service;;
    esac
done

```
Create a user services configuration:

~/.local/share/systemd/user/watch_session.service                                      

```
[Unit]
Description=Stop and restart streamdeck service on session lock.

[Service]
Type=simple
ExecStart=/usr/local/bin/watch_session.sh
Restart=on-failure

[Install]
WantedBy=default.target
```

3. Enable/start service

$ systemctl --user daemon-reload
$ systemctl --user enable watch_session.service
$ systemctl --user start watch_session.service

**Comment by shawngmc** [comment]
As an aside, if this does get integrated, please allow it to be optional (if not granular or a specific page).

I have a use case where I want to be able to use it to run specific vetted commands even while the PC is locked (specifically, controlling a USB switch and an HDMI switch via serial and/or HTTP).

**Comment by coolapso** [comment]
not going to be implemented: https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/discussions/203

**Comment by Core447** [comment]
> not going to be implemented: #203

StreamController has an option to block deck inputs when the computer is locked, which works on gnome and kde.
